MRN-Code/coinstac
# Tabular Variables with Expanded Source Type

* **Problem:**
   1. File source types might require options not covered by a generic "File" type.
   1. Bootstrap Rows/Columns get crowded with more than four columns in the UI.
* **Solution:**
   1. Add additional file source types to source list.
   1. Present variable data in table.
   1. Dynamic row appears with additional options if special source type selected.
   1. Operation and operator columns added for future data manipulation.
* **Tests:**
   1. View _Pipelines_ navigation item and select existing pipeline to see details.
   1. Expand any non-preprocessing computation in the pipeline list to see tabular variables.
   1. Add a new variable selecting **Freesurfer Data** as your data source to see dynamic options row appear.
